src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleIdInput/index.tsx

```diff
@@ -108,51 +119,68 @@ const SampleIdInput = ({
     setShowAddButton(true);
   };
 
+  const onInputBlur = () => {
+    if (!hasUnsavedChanges) {
+      setShowAddButton(false);
+    }
+  };
+
+  const onInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e?.target?.value;
+    setInputValue(value);
+    setHasUnsavedChanges(!!value);
```

Comment:
why is this `!!value` instead of `value`? Or is this some javascript syntax i'm not familiar with?

Reply:
`!!value` turns value into a boolean